src/hash_table.c

```diff
@@ -880,14 +887,28 @@ lyht_remove_with_resize_cb(struct hash_table *ht, void *val_p, uint32_t hash, va
             }
 
             /* shrink */
-            ret = lyht_resize(ht, 0);
+            ret = lyht_resize(ht, -1);
 
             if (resize_val_equal) {
                 lyht_set_cb(ht, old_val_equal);
             }
         }
     }
+    
+    r = ((ht->size - ht->used - ht->invalid) * 100) / ht->size;
+    if ((r < LYHT_REHASH_PERCENTAGE) && (ht->size > LYHT_MIN_SIZE)) {
```

Comment:
Like I said before, `(ht->size > LYHT_MIN_SIZE)` is redundant because you are not making the table smaller.

Reply:
modify